### PR TITLE
Use a temp dir for the container tarball

### PR DESCRIPTION
**Pull request by bcrochet**
For the tarball extraction, we will use a temporary directory, and
clean it up when the tests are complete.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

File: certification/internal/shell/engine.go [hunk 1]
```diff
@@ -126,14 +129,20 @@ func (e *PolicyEngine) GetContainerFromRegistry(containerLoc string, logger *log
 	}
 	lines := strings.Split(string(stdouterr), "\n")
 
+	tempdir, err := os.MkdirTemp(os.TempDir(), "preflight-*")
```
Comment (komish) [inline certification/internal/shell/engine.go:133]: Is this supposed to be `os.Mkdir`? I'm getting `os.MkdirTemp` doesn't exist

Comment (komish) [reply certification/internal/shell/engine.go:133]:
```$ GOOS=linux make build && rsync -zva preflight lab:
go build -o preflight -ldflags "-X github.com/komish/preflight/version.commit=d6396e6ed68ce10f2be22a85d87cce43e9cf6566" main.go
# github.com/komish/preflight/certification/internal/shell
certification/internal/shell/engine.go:132:18: undefined: os.MkdirTemp
make: *** [build] Error 2```

Comment (komish) [reply certification/internal/shell/engine.go:133]: Looks like this is new in 1.16. I'll update.

Comment (bcrochet) [reply certification/internal/shell/engine.go:133]: Correct. Guess I should update go.mod?

Comment (komish) [reply certification/internal/shell/engine.go:133]: @bcrochet yup, let's do that.

**Comment by komish** [comment]
@bcrochet I _believe_ I'm running into an issue extracting the tarball. From what I can tell, we're either not creating the temporary directory, or we're not saving the tarball to the right place when we extract it.

Added some log lines, which we might want to do anyway with the new logging facility.

```
time="2021-06-25T10:57:34-05:00" level=debug msg="input tarball path to extract: /tmp/preflight-192584868/272209ff0ae5fe54c119b9c32a25887e13625c9035a1599feba654aa7638262d.tar"
time="2021-06-25T10:57:34-05:00" level=error msg="unable to extract tarball contents to output directory:exit status 2"
time="2021-06-25T10:57:34-05:00" level=trace msg="tar: /tmp/preflight-192584868/272209ff0ae5fe54c119b9c32a25887e13625c9035a1599feba654aa7638262d.tar: Cannot open: No such file or directory\ntar: Error is not recoverable: exiting now\n"
time="2021-06-25T10:57:34-05:00" level=error msg="unable to extract the container: failed to extract tarball: exit status 2"
```

**Comment by bcrochet** [comment]
> @bcrochet I _believe_ I'm running into an issue extracting the tarball. From what I can tell, we're either not creating the temporary directory, or we're not saving the tarball to the right place when we extract it.
> 
> Added some log lines, which we might want to do anyway with the new logging facility.
> 
> ```
> time="2021-06-25T10:57:34-05:00" level=debug msg="input tarball path to extract: /tmp/preflight-192584868/272209ff0ae5fe54c119b9c32a25887e13625c9035a1599feba654aa7638262d.tar"
> time="2021-06-25T10:57:34-05:00" level=error msg="unable to extract tarball contents to output directory:exit status 2"
> time="2021-06-25T10:57:34-05:00" level=trace msg="tar: /tmp/preflight-192584868/272209ff0ae5fe54c119b9c32a25887e13625c9035a1599feba654aa7638262d.tar: Cannot open: No such file or directory\ntar: Error is not recoverable: exiting now\n"
> time="2021-06-25T10:57:34-05:00" level=error msg="unable to extract the container: failed to extract tarball: exit status 2"
> ```

Looks like I messed up a rebase. Fixed.

**Review by samira-barouti — APPROVED** [review]

**Pull request merged**